src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,213 @@
+
+    # Query synonyms: Mondo
+    mondo_df = pd.read_csv(mondo_synonyms_path, sep='\t').rename(columns={'?dbXref': 'source_id'})
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    # - filter by source
+    mondo_df['source_prefix'] = mondo_df['source_id'].apply(lambda x: x.split(':')[0])
+    mondo_df = mondo_df[mondo_df['source_prefix'].isin(owned_prefix_map.keys())]
+    del mondo_df['source_prefix']
+    # - filter by source previously filtered mappings_df (obsoletes & deletions)
+    mondo_df = mondo_df[mondo_df['mondo_id'].isin(set(mappings_df['mondo_id']))]  # alternative: JOIN-based solution
+    # - filter out rows which don't appear in mondo.sssom.tsv for whatever reason
+    mondo_df = mondo_df.merge(mappings_df[['mondo_id', 'source_id']], on=['mondo_id', 'source_id'], how='inner')
+    # - filter out any non-source-ID, source-specific synonym xrefs
+    #   Not future-proof, as this list has to be maintained, but as of 2024/07/15 there is only one entry, so there is
+    #   not high risk of this being a problem. Alternative: Come up w/ regex for valid codes for all sources. Will
+    #   mostly be numbers, but can include . and -. Not sure if can include alpha chars.
+    mondo_df = mondo_df[~mondo_df['source_id'].isin(NON_SOURCE_ID_SOURCE_SPECIFIC_SYNONYM_XREFS)]
+    # - add labels
+    mondo_df = mondo_df.merge(mappings_df, on=['mondo_id', 'source_id'], how='left')
+
+    # Determine synchronization cases    # -confirmed
+    confirmed_df = mondo_df.merge(source_df, on=['mondo_id', 'source_id', 'synonym_scope', 'synonym'], how='inner')
+    confirmed_df['case'] = 'confirmed'
+
+    # -updated
+    updated_df = mondo_df.merge(source_df, on=['mondo_id',  'source_id', 'synonym',], how='inner').rename(
+        columns={'synonym_scope_x': 'synonym_scope_mondo', 'synonym_scope_y': 'synonym_scope'})
+    updated_df = updated_df[updated_df['synonym_scope_mondo'] != updated_df['synonym_scope']]
+    updated_df = updated_df[[
+        'mondo_id', 'mondo_label', 'source_id', 'source_label', 'synonym_scope_mondo', 'synonym_scope', 'synonym',
+        'synonym_type']]
+    updated_df['case'] = 'updated'
+
+    # -added
+    added_df = source_df.merge(
+        mondo_df, on=['mondo_id', 'source_id', 'synonym_scope', 'synonym'], how='left', indicator=True)
+    added_df = added_df[added_df['_merge'] == 'left_only'].drop(columns=['_merge', 'mondo_label', 'source_label'])
+    # - add labels
+    for ont in ['mondo', 'source']:
+        added_df = added_df.merge(
+            mappings_df[[f'{ont}_id', f'{ont}_label']].drop_duplicates(), on=f'{ont}_id', how='left')
+    added_df = _filter_non_intersecting_rows(added_df, updated_df, ['mondo_id', 'source_id', 'synonym'])
+    del added_df['synonym_type']
+    added_df['case'] = 'added'
+
+    # -deleted
+    deleted_df = pd.DataFrame()
+    # todo: reactivate when ready. considerations: what role source_id plays in this logic
+    if not deactivate_deleted:
```

Comment:
## `-deleted` case deactivated
@twhetzel Just letting you know; feel free to mark resolved.